meteor/client/ui/ClockView/PresenterScreen.tsx

```diff
@@ -30,83 +33,157 @@ interface RundownOverviewProps {
-						startsAt: 0,
-						willProbablyAutoNext: false,
-					})
-				}),
+	const segmentIndex = orderedSegmentsAndParts.segments.findIndex((segment) => segment._id === partInstance.segmentId)
+	if (currentRundown) {
+		const showStyleBase = ShowStyleBases.findOne(showStyleBaseId)
+
+		if (showStyleBase) {
+			// This registers a reactive dependency on infinites-capping pieces, so that the segment can be
+			// re-evaluated when a piece like that appears.
+
+			let o = RundownUtils.getResolvedSegment(
+				showStyleBase,
+				playlist,
+				orderedSegmentsAndParts.segments[segmentIndex],
+				new Set(orderedSegmentsAndParts.segments.map((segment) => segment._id).slice(0, segmentIndex)),
+				orderedSegmentsAndParts.parts.map((part) => part._id),
+				true,
+				true
+			)
+
+			segment = extendMandadory<DBSegment, SegmentUi>(o.segmentExtended, {
+				items: o.parts,
 			})
+
+			partInstanceUi = o.parts.find((part) => part.instance._id === partInstance._id)
+		}
+	}
+
+	return {
+		showStyleBaseId: showStyleBaseId,
+		segment: segment,
+		partInstance: partInstanceUi,
+	}
+}
+
+export const getPresenterScreenReactive = (props: RundownOverviewProps): RundownOverviewTrackedProps => {
+	let playlist: RundownPlaylist | undefined
+	if (props.playlistId)
+		playlist = RundownPlaylists.findOne(props.playlistId, {
+			fields: {
+				lastIncorrectPartPlaybackReported: 0,
+				modified: 0,
+				nextPartManual: 0,
+				previousPersistentState: 0,
+				rundownRanksAreSetInSofie: 0,
+				trackedAbSessions: 0,
+				restoredFromSnapshotId: 0,
+			},
 		})
+	let segments: Array<SegmentUi> = []
+	let showStyleBaseIds: ShowStyleBaseId[] = []
+	let rundowns: { [key: string]: Rundown } = {}
+	let rundownIds: RundownId[] = []
+
+	let currentSegment: SegmentUi | undefined = undefined
+	let currentPartInstanceUi: PartUi | undefined = undefined
+	let currentShowStyleBaseId: ShowStyleBaseId | undefined = undefined
+
+	let nextSegment: SegmentUi | undefined = undefined
+	let nextPartInstanceUi: PartUi | undefined = undefined
+	let nextShowStyleBaseId: ShowStyleBaseId | undefined = undefined
 
-		if (playlist.currentPartInstanceId) {
-			const { currentPartInstance, nextPartInstance } = playlist.getSelectedPartInstances()
-			const partInstance = currentPartInstance || nextPartInstance
-			if (partInstance) {
-				const currentRundown = Rundowns.findOne(partInstance.rundownId)
-				if (currentRundown) {
-					showStyleBaseId = currentRundown.showStyleBaseId
-				}
+	if (playlist) {
+		rundowns = playlist.getRundownsMap()
+		const orderedSegmentsAndParts = playlist.getSegmentsAndPartsSync()
+		rundownIds = protectStringArray(Object.keys(rundowns))
+		showStyleBaseIds = Object.values(rundowns).map((rundown) => rundown.showStyleBaseId)
+		const { currentPartInstance, nextPartInstance } = playlist.getSelectedPartInstances()
+		const partInstance = currentPartInstance || nextPartInstance
+		if (partInstance) {
+			const infinitesEndingPieces = PieceInstances.find({
```

Comment:
This constant doesn't seem to be used anywhere, does it have to be assigned?

Reply:
No, it probably doesn't - good find.